xwiki-pro-macros-api/src/main/java/com/xwiki/macros/userlist/internal/macro/UserListMacro.java

```diff
@@ -19,40 +19,47 @@
  */
 package com.xwiki.macros.userlist.internal.macro;
 
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Singleton;
-import javax.script.ScriptContext;
 
 import org.xwiki.component.annotation.Component;
+import org.xwiki.displayer.HTMLDisplayerException;
+import org.xwiki.displayer.HTMLDisplayerManager;
+import org.xwiki.model.reference.EntityReferenceSerializer;
 import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.RawBlock;
 import org.xwiki.rendering.macro.AbstractMacro;
 import org.xwiki.rendering.macro.MacroExecutionException;
+import org.xwiki.rendering.syntax.Syntax;
 import org.xwiki.rendering.transformation.MacroTransformationContext;
-import org.xwiki.script.ScriptContextManager;
-import org.xwiki.template.Template;
-import org.xwiki.template.TemplateManager;
+import org.xwiki.text.StringUtils;
 
 import com.xwiki.macros.userlist.macro.UserListMacroParameters;
+import com.xwiki.macros.userlist.macro.UserReferenceList;
 
 /**
  * This macro displays a list of users with their name and avatar.
  *
  * @version $Id$
  */
-
 @Component
 @Named("userList")
 @Singleton
 public class UserListMacro extends AbstractMacro<UserListMacroParameters>
 {
     @Inject
-    private TemplateManager templateManager;
+    private HTMLDisplayerManager htmlDisplayerManager;
 
     @Inject
-    private ScriptContextManager scriptContextManager;
+    @Named("default")
```

Comment:
You should just omit the ``@Named`` annotation when the hint is "default" (your code works, but it can be simplified).

Reply:
Right, done in latest commit